### ISSUE TITLE
Using -s option for reference file

### ISSUE THREAD

**Issue by hm22rice** [barricklab/breseq#301]
Hello,
I am trying to understand the function of the -s option and if it will be applicable for me.

I am using a reference strain "A" whose complete genome is available on Genbank. I have modified strain "A" by chromosomally inserting a GFP gene in it. However, I do not know the insertion site of GFP. 

If I want to use breseq to call the junction created by the GFP gene in my reference, would it be suitable to use the -s option? If so, would this usage be correct:
breseq -r   A.gbk   -s   GFP.fasta   short_reads.fq.gz

Once I have identified the insertion site, I would like to create a modified reference sequence which includes the insertion. Would I be able to use gdtools to do that? Or is there another way to do so.

Thank you so much,
Heer

**Comment by hm22rice** [comment]
Hi Dan,
Thank you so much for the quick response. I will give this a try and let you know if I have any other questions. 
I appreciate your help.

Heer

**Comment by jeffreybarrick** [comment]
What @danieldeatherage said, and some other info that might be helpful to you or someone else who comes across this...

1/ There's also an `INT` mutation type that you can use for copying something over from one sequence to another with `gdtools APPLY`. It's more powerful than an `INS` for this case. It will let you reverse complement the copy if it is on the other strand and copy over the CDS/gene annotations if you used an annotated GFP sequence file that was in GenBank/GFF3 format.

The line would read something like:
```
INT<TAB>.<TAB>.<TAB><Seq ID><TAB><position where GFP was inserted><tab>GFP:1-700
```
If the GFP sequence was 700 bases long, names GFP *in* `GFP.fasta` and in the forward direction. `GFP:700-1` if it was in the reverse direction.

More info:
https://barricklab.org/twiki/pub/Lab/ToolsBacterialGenomeResequencing/documentation/gd_format.html#int-integration-mutation

2/ If you inserted more than just GFP (some flanking sequences or it's in a transposon), be sure that you make the GFP sequence go all the way up to those ends that you know. _breseq_ will have trouble finding junctions if your sequence and the genome sequence don't "meet" or at least come close (within something like <20 base pairs).

**Comment by hm22rice** [comment]
Thank you so much for the help. I was able to integrate the exogenous sequence into the genome of strain A. 

Just a couple of things I did differently (putting this out here for anybody else who might want to do something similar):

1. I used the INT mutation type in the new genome diff file. The line needed 7 tab separated columns instead of 6 so I modified it to look like this:
   INT<TAB>.<TAB>.<TAB><Seq ID><TAB><position where GFP was inserted><tab><number of bases to replace in reference genome><tab>GFP:1-700

In my case, number of bases to replace in reference genome = 0

2. After creating the new genome diff file, I ran gdtools using the following command line:
gdtools APPLY -r A.gbk -r GFP.fasta -f gff3 -o A.updated.gff3 new_file.gd

I have two reference sequences since the new genome diff file only has the nucleotide positions of the integrated sequence but not the actual sequence. 

Thank you,
Heer
